### PR TITLE
Forms

### PR DESCRIPTION
**Pull request by joyapisi**
## Work Done:
- Create a method called current_user in my ApplicationController to make current user data available to all controllers.
   - It returns the first user from the database.
- Create forms to perform the following functions:
   - Create a Post on behalf of the current_user (use the method that you created in your ApplicationController).
   - Create a comment on behalf of the current_user (use the method that you created in your ApplicationController).
   - Allow Users to add likes to Posts.

### CODE REVIEW

File: app/views/shared/_users.html.erb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<div id="each-user">
+    <div class="boxes img-card">
+      <img src=<%= user.photo %> alt="profile" class="img-card">
+    </div>
+    <div class=" boxes user-card" >
+      <a href="/users/<%= user.id%>"> <h2 class="user-name"> <%= user.name %></h2></a>
+      <p class="post-counter">Number of posts: <%= user.update_user_posts_counter%> </p>
+    </div>
+ 
+</div>
```
Comment (anagudelogu) [inline app/views/shared/_users.html.erb:10]: - [ ] **[OPTIONAL]** Please consider taking a look at your Posts counter! Currently, when I create a new post, it doesn't get updated. 🤓 🔧 

image

File: app/views/posts/index.html.erb [hunk 2]
```diff
@@ -1 +1,21 @@
...
+<% Post.recent_comments(post).each  do |comment|%>
+    <p><%= comment.author.name%> - <%= comment.text %></p>
+    <%end%>
+</div>
+<br />
+<%end%>
+
+<div class="pagination-btn">
+<button>Pagination</button>
+</div>
```
Comment (anagudelogu) [inline app/views/posts/index.html.erb:21]: - [ ] **[OPTIONAL]** Also, please make sure that your `Post#index` view displays the correct information, currently there are two posts but none of them appear in this view. 🤓 🔧 

image

**Review by anagudelogu — APPROVED** [review]
# STATUS: APPROVED! 🍾 🎆 🥇 

Hi @joyapisi, 
I'm Andres.

You did a great job working on this milestone, your project is complete! 🚀 
There is nothing else to say other than... it's time to merge it 🤖  
Congratulations! 🎉 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Descriptive PR title, summary and commit messages. 💬 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/posts/index.html.erb (hunk 2 above)
Comment (joyapisi) [reply app/views/posts/index.html.erb:21]: Okay. Thank you!I will work on that.

**Pull request merged**